### ISSUE TITLE
SwerveWithPathPlanner race condition causes incorrect path replanning

### ISSUE THREAD

**Issue by agasser** [CrossTheRoadElec/Phoenix6-Examples#27]
See the attached video running SwerveWithPathPlanner in the simulator. I had similar results running a real robot.

**Background**
This is running the example with two modifications:
1) Updated PathPlanner to 2024 beta 6.1
2) Changed `getAutomonousCommand()` to return the PathPlanner auto:
```java
  public Command getAutonomousCommand() {
    return AutoBuilder.buildAuto("Tests");
  }
```

**Expected Behavior**
1) The robot's pose should be updated from the default (0, 0) to the starting pose of the path.
2) The robot should follow the path. The path drives a short arc above the charge station.

**Actual Behavior**
The robot's pose is properly updated to the path starting pose. However, the path is replanned as though the starting pose was not updated.

https://github.com/CrossTheRoadElec/Phoenix6-Examples/assets/1993518/a7a7cbe7-7d58-480e-b309-0f28d2c5d0dc

https://github.com/CrossTheRoadElec/Phoenix6-Examples/assets/1993518/ec11191b-fcd9-421a-97ee-7fe39948572e

**Root Cause**
I believe the root cause is because the `SwerveDrivetrain.seedFieldRelative(Pose2d)` method updates the `SwerveDrivePoseEstimator`, but it does not update `m_cachedState`. The pose supplier for PathPlanner is `()->this.getState().Pose`.
When PathPlanner starts running auto, it calls `SwerveDrivetrain.seedFieldRelative(Pose2d)` to update the robot's pose. Then it calls `SwerveDrivetrain.getState().Pose` to get the current pose and `getState()` returns the pose from before it was updated by PathPlanner.
In rare cases, this race condition yields the correct results because telemetry gets updated between calls to `seedFieldRelative` and `getState`.

**Workaround**
I found two possible workarounds
1) Change PathPlanner's `ReplanningConfig` so `enableInitialReplanning` is `false`. This works, but it's not really "correct."
2) Override `seedFieldRelative(Pose2d)` to update the cached pose:
```java
    @Override
    public void seedFieldRelative(Pose2d location) {
        try {
            m_stateLock.writeLock().lock();
            m_odometry.resetPosition(Rotation2d.fromDegrees(m_yawGetter.getValue()), m_modulePositions, location);
            m_cachedState.Pose = location;
        } finally {
            m_stateLock.writeLock().unlock();
        }
    }
```

**Comment by agasser** [comment]
Here's a more future proof/resistant workaround:
```java
  @Override
  public void seedFieldRelative(Pose2d location) {
    m_stateLock.writeLock().lock();
    try {
      m_cachedState.Pose = location; // Workaround: update cached state
      super.seedFieldRelative(location);
    } finally {
      m_stateLock.writeLock().unlock();
    }
  }
```

**Comment by bhall-ctre** [comment]
This is fixed in the 24.1.0 release
